### PR TITLE
feat(metaflags): changing storybook.json into metaflags.json

### PR DESCRIPTION
**Pull request by pvinis**
The type of this PR is: **Feature**

This PR resolves []

### Description

This PR comes from some changes we did for the new app shell. Trickling down small improvements to main, while we are in a separate branch :)

The idea behind this PR is that the same way we controlled storybook initialization, we can use it for more things. More specifically, a "new app shell" flag. There is also the usecase of the loggers that @patrinoua is working on #6099. I'm sure we will find more uses for this.

The way it works is:
- the first time you pull after this PR is merged, remember to `yarn setup:artsy`, to do some cleanup.
- from there, you can see the `metaflags.json` file that you will have locally. its ignored, local only, but it controls things for you!
- for now its just for storybook, but soon it will be for at least two other useful things (app shell and loggers)
- the `index.ios.js` and `index.android.js` files, both check for this json file, and control things for us. in general, when we use it for other things, like loggers and app shell, we can quickly import that json file and get a boolean that can give us some control on dev things.
- this file will always be restored with `yarn setup:artsy`, in case something is broken or non-normal. it will be restored from the `metaflags.example.json` file.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- metaflags! - pavlos

</details>

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (metaflags! - pavlos)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7903cd72300ed8012817ccde3be44a959f815ee2
</p>

File: index.android.js [hunk 1]
```diff
@@ -1,17 +1,17 @@
 global.__TEST__ = false
 global.__STORYBOOK__ = false
 
-// start storybook depending on content of storybook.json
-let startStorybook = false
-
+let metaflags = {
+  startStorybook: false,
+}
```
Comment (olerichter00) [inline index.android.js:6]: Let's add some documentation to explain what a meta flag is and how we are using it.

File: scripts/set-storybook-environment.js [hunk 2]
```diff
@@ -1,23 +1,17 @@
 #!/usr/bin/env node
 
-// Create or change `storybook.json` depending on STORYBOOK env variable
+// Change `metaflags.json` depending on STORYBOOK env variable
```
Comment (olerichter00) [inline scripts/set-storybook-environment.js:3]:
```suggestion
// Change `metaflags.json` depending on env variables (e.g. `STORYBOOK`)
```

**Review by olerichter00 — CHANGES_REQUESTED** [review]

File: index.android.js (hunk 1 above)
Comment (pvinis) [reply index.android.js:6]: yup yup! adding it now.

Comment (pvinis) [reply index.android.js:7]: 👍

**Review by olerichter00 — APPROVED** [review]
Thanks for the docs 🥇

**Comment by pvinis** [comment]
this should be ready too 👍

**Comment by pvinis** [comment]
lets go

**Pull request merged**